shared/mail/mail.go

```diff
@@ -208,10 +208,6 @@ func NewSMTPClient(ctx context.Context, conn net.Conn, config *SMTPConfig) (*smt
 }
 
 func TestConnection(config *SMTPConfig) error {
```

Comment:
I think this function should not be called if `SendEmailNotifications` is not true at all. You can check that setting on the call site here:

https://github.com/mattermost/mattermost-server/blob/6ab7cd0f1a7fd8b4540830d5fdef6c69e0d14510/app/server.go#L1199